packages/core/i18n/src/i18n.ts

```diff
@@ -52,19 +52,32 @@ export const createI18n = <MessageSource extends Record<string, any>>
   const { $t, ...otherProps } = intlOriginal
   const intl = otherProps
 
+  /**
+   * Shamefully normalize a timestamp to be in seconds as some APIs are returning timestamps in milliseconds
+   * TODO: Remove this function once all timestamps are normalized from the backend
+   * @param {number} timestamp a unix timestamp in seconds *or milliseconds*
+   * @returns {number} a unix timestamp in seconds
+   */
+  const shamefullyNormalizeTimeStamp = (timestamp: number): number => {
+    if (timestamp.toString().length === 13) {
+      return Math.floor(timestamp / 1000)
+    }
+    return timestamp
+  }
+
   /**
    * Formats a unix timestamp into a formatted date string
    * @param {Number} timestamp a unix timestamp in seconds
    * @returns a date string formatted like 'Apr 6, 2022 10:50'
    */
-  const formatUnixTimeStamp = (timestamp: number): string => {
+  const formatUnixTimeStamp = (timestamp?: number | null): string => {
     const invalidDate = 'Invalid Date'
     if (!timestamp) {
-      return invalidDate
+      return '-'
```

Comment:
```suggestion
      return '—'
```

Can we use emdash instead please?

Reply:
I think I should just remove this. We are already handling `null` before this.